### DISCUSSION TITLE
Example of using generated by plugin client code

### DISCUSSION THREAD

**Discussion by chemeugene** [kobylynskyi/graphql-java-codegen#609] [Q&A] [answered]
Hi!

I'm looking for some examples of code how to call GraphQL API from kotlin. 
I used graphql-java-codegen with generateClient=true option. But how to use generated requests, response, projections? 
Is there any reference example? 

Thanks in advance

**Comment by kobylynskyi** [comment, marked as answer]
Hey Eugene
Feel free to check following modules:
- plugins/gradle/example-client
- plugins/gradle/example-client-kotlin 
- plugins/maven/example-client

In particular, you can take a look at the following class to see how request/response classes are being used: https://github.com/kobylynskyi/graphql-java-codegen/blob/master/plugins/gradle/example-client/src/main/java/io/github/kobylynskyi/order/external/product/ProductServiceGraphQLClient.java

**Comment by jxnu-liguobin** [comment]
This is a Github client example for kotlin/scala/java https://github.com/jxnu-liguobin/github-graphql-client/blob/master/src/main/kotlin/io/github/graphql/example/KotlinClientExample.kt implemnet by dynamic proxy.

A simpler example of kotlin: https://github.com/kobylynskyi/graphql-java-codegen/blob/master/plugins/gradle/example-client-kotlin/src/main/kotlin/io/github/dreamylost/QueryResolverImpl.kt
